packages/swapper/src/swappers/ButterSwap/xhr.ts

```diff
@@ -0,0 +1,214 @@
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+
+import type { SwapErrorRight } from '../../types'
+import { TradeQuoteError } from '../../types'
+import { makeSwapErrorRight } from '../../utils'
+import type {
+  BridgeInfoApiResponse,
+  BuildTxResponse,
+  FindTokenResponse,
+  RouteAndSwapResponse,
+  RouteResponse,
+  SupportedChainListResponse,
+} from './types'
+import { butterHistoryService } from './utils/butterSwapHistoryService'
+import { butterService } from './utils/butterSwapService'
+
+type ButterSwapPromise<T> = Promise<Result<T, SwapErrorRight>>
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-supportedchainlist
+ */
+export const getSupportedChainList = async (): ButterSwapPromise<SupportedChainListResponse> => {
+  const result = await butterService.get<SupportedChainListResponse>('/supportedChainList')
+  if (result.isErr()) return Err(result.unwrapErr())
+  // No runtime validation, just return the data
+  return Ok(result.unwrap().data)
+}
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-findtoken
+ */
+export const findToken = async (
+  chainId: number,
+  address: string,
+): ButterSwapPromise<FindTokenResponse> => {
+  const result = await butterService.get<FindTokenResponse>('/findToken', {
+    params: { chainId, address },
+  })
+  if (result.isErr()) return Err(result.unwrapErr())
+  return Ok(result.unwrap().data)
+}
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-route
+ */
+export const getRoute = async (
```

Comment:
nitpick: butter terminology across all these 

```suggestion
export const getButterRoute = async (
```

Reply:
Or actually can we make this `getRate` or `getButterRate` ? Then `buildTx` will be `getQuote` / `getButterQuote` which will read a lot better than current

Reply:
https://github.com/shapeshift/web/pull/9799/commits/e84c9b2afe2186747437e9f4b302fd02a35857b0